src/main/java/lotto/domain/Division.java

```diff
@@ -33,25 +38,23 @@ public int getPrize() {
     }
 
     public static Division valueOf(int matchCount) {
-        if (matchCount == 6) {
-            return FIRST;
-        }
-        if (matchCount == 5) {
-            return THIRD;
-        }
-        if (matchCount == 4) {
-            return FOURTH;
-        }
-        if (matchCount == 3) {
-            return FIFTH;
-        }
-        if (matchCount == 0 || matchCount == 1 || matchCount == 2) {
-            return LOSE;
-        }
-        throw new IllegalArgumentException("로또 맞은 번호의 개수가 6개 초과입니다.");
+        return Division.valueOf(matchCount, false);
     }
 
-    public final static Map<Division, Integer> getMatchCounts() {
-        return matchCounts;
+    public static Division valueOf(int matchCount, boolean matchBonus) {
+        return Arrays.stream(values())
+                .filter(division -> division.matchCounts.contains(matchCount))
+                .filter(division -> {
+                    if (matchCount == 5) {
+                        return division.matchBonus == matchBonus;
+                    }
+                    return true;
+                })
```

Comment:
```suggestion
                .filter(division -> division.matchBonus == matchBonus)
```
5일때를 특정하지 않아도 stream처리가되면서 자연스럽게 필터링이 될 것 같네요 😄